### PR TITLE
Allow passing options to rebuildServer

### LINKED ISSUE #414: [Rackspace] Rebuild Server - cannot send flavorRef or other optional parameters
According to the Rackspace docs when making the call to rebuild a server a flavorRef is required (there are also optional properties)
http://docs.rackspace.com/servers/api/v2/cs-devguide/content/Rebuild_Server-d1e3538.html

The rebuild command on the compute client does not allow you to send the flavorRef or any of the optional properties 

Despite not setting a flavorRef the rebuild request works but I cannot script rebuilding to a new flavor, nor can I send any of the optional parameters.

### PR DESCRIPTION
**Pull request by ycombinator**
Fixes #414

### CODE REVIEW

File: lib/pkgcloud/openstack/compute/client/servers.js [hunk 1]
```diff
@@ -275,18 +275,44 @@ exports.rebootServer = function rebootServer(server, options, callback) {
...
+ * @param {array}           options.personality  Personality files - path and contents
+ * @param {String}          options.personality[n].path      Personality file path
+ * @param {String}          options.personality[n].contents  Personality file contents
+ * @param {String}          options['OS-DCF:diskConfig']    The disk configuration value ("AUTO" | "MANUAL").
  * @param {Function}        callback
  * @returns {*}
  */
-exports.rebuildServer = function rebootServer(server, image, callback) {
-  var imageId = image instanceof base.Image ? image.id : image;
+exports.rebuildServer = function rebootServer(server, options, callback) {
```
Comment (kenperkins) [inline lib/pkgcloud/openstack/compute/client/servers.js]: Ah. I like this. I was thinking of just replacing `server` and `image` with `options`, but this makes more sense. Well done.

File: test/openstack/compute/client/rebuildServer-test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      },
+      function (next) {
+        authServer.close(next);
+      }
+    ], done);
+  });
+});
+
+setupRebuildServerWithImageIdMock = function(servers) {
+		servers.authServer
```
Comment (kenperkins) [inline test/openstack/compute/client/rebuildServer-test.js]: Indentation seems a bit off here :)

Comment (ycombinator) [inline test/openstack/compute/client/rebuildServer-test.js]: LOL yeah. Fixing.

**Comment by ycombinator** [comment]
@kenperkins ready for review again.

**Comment by kenperkins** [comment]
lgtm

**Pull request merged**